core/server/src/main/java/alluxio/master/journal/JournalWriter.java

```diff
@@ -179,16 +213,15 @@ private synchronized OutputStream openCurrentLog() throws IOException {
    */
   private synchronized void deleteCompletedLogs() throws IOException {
     LOG.info("Deleting all completed log files...");
-    // Loop over all complete logs starting from the beginning.
-    // TODO(gpang): should the deletes start from the end?
```

Comment:
this method finds the logs by seeing how high it can count before the log of that number doesn't exist. If we have 10 logs and only delete log #1, the next time we try to delete the logs we won't delete logs 2-10 because we see that log 1 doesn't exist.